ts/src/index.ts

```diff
@@ -117,12 +121,26 @@ export class AutoPush {
                   this.addCacheHeaders(headers, stats);
                 },
                 onError: (err) => {
-                  console.error(err);
                   pushStream.end();
                   reject(err);
                 },
               });
-        });
+        };
+        // Node 9.4.0 changed the callback function signature.
+        let callback: PushStreamCallback;
+        if (isNode940) {
```

Comment:
I expect that the offending change from 9.4.0 is going to be backported to Node 8 as well. For that version of Node 8 this code will break.

An alternative to having a semver check in the code would be to use a `engines` field in package.json that requires Node 9.4+ or Node 8.10+ (not yet released)

Reply:
Oh, is it true that the same change will be backported to 8? If we limit the engines to 9.4+ and 8.10+, then it'll be broken with current Node 8...

Reply:
Does the "engines" field actually work as expected?

I added
```json
  "engines": {
    "node": ">=9.4.0 || >8.9.4 <9"
  },
```

But when I do `npm install -g` with Node 8.9.4, it doesn't emit any warnings.

Reply:
What's that <9 in that expression, not sure if I understand the expression.

Reply:
"version >= 9.4.0 OR (version > 8.9.4 AND < 9.0.0)"

Am I missing something?

Reply:
Ah, whitespace does AND.

Reply:
Not sure why it doesn't warn though.

Reply:
https://runkit.com/embed/7g1z4mkyclcy